Plogon/BuildProcessor.cs

```diff
@@ -149,19 +160,20 @@ async Task GetDependency(string name, NugetLockfile.Dependency dependency, Direc
         await File.WriteAllBytesAsync(depPath, data);
     }
 
-    private async Task RestorePackages(DirectoryInfo pkgFolder, FileInfo lockFile, HttpClient client)
+    private async Task RestorePackages(DirectoryInfo pkgFolder, FileInfo lockFile)
     {
         var lockFileData = JsonConvert.DeserializeObject<NugetLockfile>(File.ReadAllText(lockFile.FullName));
 
         if (lockFileData.Version != 1)
             throw new Exception($"Unknown lockfile version: {lockFileData.Version}");
 
-        var runtime = lockFileData.Runtimes.First();
-        Log.Information("Getting packages for runtime {Runtime}", runtime.Key);
-
-        foreach (var dependency in runtime.Value.Where(x => x.Value.Type != NugetLockfile.Dependency.DependencyType.Project))
+        foreach (var runtime in lockFileData.Runtimes)
         {
-            await GetDependency(dependency.Key, dependency.Value, pkgFolder, client);
+            Log.Information("Getting packages for runtime {Runtime}", runtime.Key);
+            
+            await Task.WhenAll(runtime.Value
+                .Where(x => x.Value.Type != NugetLockfile.Dependency.DependencyType.Project)
+                .Select(dependency => GetDependency(dependency.Key, dependency.Value, pkgFolder, new HttpClient())).ToList());
```

Comment:
Why are you not reusing the HttpClient? It'll keep the connection to the server open if you do

Reply:
Was too lazy to look up threading guarantees when you GetAsync on the same client from different threads. If you know that its fine I'll add it back in a sec.

Reply:
should be good

Reply:
Done.